libraries/AP_Winch/AP_Winch_Daiwa.cpp

```diff
@@ -0,0 +1,227 @@
+                switch (parse_state) {
+                case ParseState::WAITING_FOR_TIME:
+                    intermediate.time_ms = (uint32_t)ret;
+                    parse_state = ParseState::WAITING_FOR_SPOOL;
+                    break;
+                case ParseState::WAITING_FOR_SPOOL:
+                    intermediate.line_length = (int32_t)ret * line_length_correction_factor;
+                    parse_state = ParseState::WAITING_FOR_TENSION1;
+                    break;
+                case ParseState::WAITING_FOR_TENSION1:
+                    intermediate.tension_uncorrected = (uint16_t)ret;
+                    parse_state = ParseState::WAITING_FOR_TENSION2;
+                    break;
+                case ParseState::WAITING_FOR_TENSION2:
+                    intermediate.tension_corrected = (uint16_t)ret;
+                    parse_state = ParseState::WAITING_FOR_THREAD_END;
+                    break;
+                case ParseState::WAITING_FOR_THREAD_END:
+                    intermediate.thread_end = (ret > 0);
+                    parse_state = ParseState::WAITING_FOR_MOVING;
+                    break;
+                case ParseState::WAITING_FOR_MOVING:
+                    intermediate.moving = constrain_int16(ret, 0, UINT8_MAX);
+                    parse_state = ParseState::WAITING_FOR_CLUTCH;
+                    break;
+                case ParseState::WAITING_FOR_CLUTCH:
+                    intermediate.clutch = constrain_int16(ret, 0, UINT8_MAX);
+                    parse_state = ParseState::WAITING_FOR_SPEED;
+                    break;
+                case ParseState::WAITING_FOR_SPEED:
+                    intermediate.speed_pct = constrain_int16(ret, 0, UINT8_MAX);
+                    parse_state = ParseState::WAITING_FOR_VOLTAGE;
+                    break;
+                case ParseState::WAITING_FOR_VOLTAGE:
+                    intermediate.voltage = (float)ret * 0.1f;
+                    parse_state = ParseState::WAITING_FOR_CURRENT;
+                    break;
+                case ParseState::WAITING_FOR_CURRENT:
+                    intermediate.current = (float)ret * 0.1f;
+                    parse_state = ParseState::WAITING_FOR_PCB_TEMP;
+                    break;
+                case ParseState::WAITING_FOR_PCB_TEMP:
+                    intermediate.pcb_temp = (float)ret * 0.1f;
+                    parse_state = ParseState::WAITING_FOR_MOTOR_TEMP;
+                    break;
+                case ParseState::WAITING_FOR_MOTOR_TEMP:
+                    intermediate.motor_temp = (float)ret * 0.1f;
+                    // successfully parsed a complete message
+                    latest = intermediate;
+                    data_update_ms = AP_HAL::millis();
+                    parse_state = ParseState::WAITING_FOR_TIME;
+                    break;
+                }
+            }
+        } else {
+            // carriage return or unexpected characters
```

Comment:
do you mean "line feed" here?  It's the only way I can see resync happening in the code as it is.

CR is handled above.

I'm *guessing* the lines are crlf terminated?

Reply:
yes, lines are terminated with crlf.  comment fixed, txs.